admin/src/pages/Reports/components/ReportsTableRow/index.tsx

```diff
@@ -0,0 +1,351 @@
+      }
+    }
+  };
+
+  const handleBlockItemClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      blockItemMutation.mutate(id);
+    }
+  };
+
+  const handleUnblockClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      unblockItemMutation.mutate(id);
+    }
+  };
+
+  const handleBlockItemThreadClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      blockItemThreadMutation.mutate(id);
+    }
+  };
+
+  const handleUnblockThreadClick = async (id) => {
+    if (canModerate) {
+      lockApp();
+      unblockItemThreadMutation.mutate(id);
+    }
+  };
+
+  const isLoading =
+    blockItemMutation.isLoading || blockItemThreadMutation.isLoading;
+  const openReports = reportedReports?.filter((_) => !_.resolved);
+  const hasReports = !isEmpty(openReports);
+  const reviewFlowEnabled =
+    canAccessReports && hasReports && !(item.blocked || item.blockedThread);
+
+  const handleClick = () => onClick();
+  const handleBlockButtonsStateChange = (disabled) =>
+    setBlockButtonsDisabled(disabled);
+
+  const renderStatus = (props) => {
+    const status = resolveReportStatus({...props});
+    const color = resolveReportStatusColor(status);
+
+    return (
+      <StatusBadge
+        backgroundColor={`${color}100`}
+        textColor={`${color}700`}
+        color={color}>
+        {getMessage(
+          {
+            id: `page.common.item.status.${status}`,
+            props: {
+              count: openReports.length,
+            },
+          },
+          status,
+        )}
+      </StatusBadge>
+    );
+  };
+
+  const renderEntryUrl = (entry) => {
+    entry ? `/content-manager/collectionType/${entry.uid}/${entry.id}` : null;
+  };
+  const renderDetailsUrl = (entry) => getUrl(`discover/${entry.id}`);
+
+  const gotApprovalFlow = !isNil(item.approvalStatus);
+  const needsApproval = gotApprovalFlow && item.approvalStatus === 'PENDING';
+
+  let actionItemsCount = 1;
+  if (reviewFlowEnabled || needsApproval) {
+    actionItemsCount = 2;
+  }
+
+  return (
+    <Tr key={item.id}>
+      <Td>
+        <BaseCheckbox
+          aria-label={getMessage(
+            'page.details.panel.discussion.warnings.reports.dialog.select',
+          )}
+          value={isChecked(item.id)}
+          // disabled={entry.resolved}
+          onValueChange={(value) => handleItemSelectionChange(item.id, value)}
+        />
+      </Td>
+      <Td>
+        <Typography textColor='neutral800' fontWeight='bold'>
+          #{item.id}
+        </Typography>
+      </Td>
+      <Td>
+        <Typography textColor='neutral800' variant='pi'>
+          {item.reason || getMessage('compontents.reason.unknown')}
+        </Typography>
+      </Td>
+      <Td style={{maxWidth: '30vw'}}>
+        <Typography textColor='neutral800' ellipsis>
+          {item.content || getMessage('compontents.content.unknown')}
+        </Typography>
+      </Td>
+      <Td style={{maxWidth: '30vw'}}>
```

Comment:
Consider extracting to `const` in module scope

Reply:
What exactly You suggest to extract?